doc: helm : tiny improvements and fixes (#3595)
operator-framework/operator-sdk: website/content/en/docs/building-operators/helm/migration.md

---
title: Migrating Helm-based Legacy Projects
linkTitle: Migrating Legacy Projects

> weight: 3

Reviewer: I intentionally made these weights separated by 100 so that we can add new pages anywhere in the list without changing the others' weights.

I think revert this back to 300.

> description: This guide walks through an example of migrating a simple nginx-operator which was built by following the [legacy quick-start][quickstart-legacy] to the new Kubebuilder-style layout.

Reviewer: This doc is pretty generic and doesn't really use the legacy sample project as an example.

Let's keep the description as is:
```suggestion
description: Instructions for migrating a legacy Helm-based project to use the new Kubebuilder-style layout.
```

weight: 3
description: This guide walks through an example of migrating a simple nginx-operator which was built by following the [legacy quick-start][quickstart-legacy] to the new Kubebuilder-style layout.
---

> This guide walks through an example of migrating a simple nginx-operator which was built by following the [legacy quick-start][quickstart-legacy] to the new layout.

Reviewer: +1 on removing this line.

description: Instructions for migrating a legacy Helm-based project to use the new Kubebuilder-style layout.
---

## Overview

The motivations for the new layout are related to bringing more flexibility to users and 
part of the process to [Integrating Kubebuilder and Operator SDK][integration-doc].

### What was changed
 
The `deploy` directory was replaced with the `config` directory including a new layout of Kubernetes manifests files:
- CRD manifests in `deploy/crds/` are now in `config/crd/bases`
- CR manifests in `deploy/crds/` are now in `config/samples`
- Controller manifest `deploy/operator.yaml` is now in `config/manager/manager.yaml` 
- RBAC manifests in `deploy` are now in `config/rbac/`

> `build/Dockerfile` is moved to `Dockerfile` in the project root directory

Reviewer: Move this to the bullet list.

Author: it is not in the bullet: 👍 
<img width="767" alt="Screenshot 2020-07-30 at 14 20 16" src="https://user-images.githubusercontent.com/7708031/88927789-dbdb3300-d26f-11ea-9647-b663c1919f56.png">

Author: Done as:

<img width="881" alt="Screenshot 2020-07-30 at 14 41 50" src="https://user-images.githubusercontent.com/7708031/88943459-6aa57b00-d283-11ea-96ec-8f56cbdff4c9.png">

### What was changed
 

> The `deploy` directory was replaced with the `config` directory including a new layout of Kubernetes manifests files

Reviewer: ```suggestion
The `deploy` directory was replaced with the `config` directory including a new layout of Kubernetes manifests files:
```

- CRD manifests in `deploy/crds/` are now in `config/crd/bases`
- CR manifests in `deploy/crds/` are now in `config/samples`

> - Controller manifest `deploy/operator.yaml` was replaced for `config/manager/manager.yaml` 

Reviewer: ```suggestion
- Controller manifest `deploy/operator.yaml` is now in `config/manager/manager.yaml` 
```

> - RBCA manifests in `deploy` are in `config/rbac/`

Reviewer: ```suggestion
- RBAC manifests in `deploy` are in `config/rbac/`
```

> The `build/Dockerfile` directory was replaced by the `Dockerfile` in the root directory

Reviewer: Move this into the bullet list.

> 
> The `build/Dockerfile` directory was replaced by the `Dockerfile` in the root directory

Reviewer: ```suggestion
- `build/Dockerfile` is moved to `Dockerfile` in the project root directory
```

So let's create a new project with the same domain (`example.com`):

```sh
$ mkdir nginx-operator
$ cd nginx-operator

> $ operator-sdk init --plugins=helm --domain=example.com

Reviewer 2: +1 to this. I've mentioned this in the [style guide issue](https://github.com/operator-framework/operator-sdk/issues/3615)

For each API in the existing project, run:
```sh
operator-sdk create api \
    --group=cache \

>     --version=<apiVersion> \

Reviewer: ```suggestion
    --version=<version> \
```
